--- bzip2-1.0.4/bzgrep.orig	2007-01-03 03:00:55.000000000 +0100
+++ bzip2-1.0.4/bzgrep	2007-01-12 14:30:06.518134000 +0100
@@ -63,9 +63,7 @@
     bzip2 -cdfq "$i" | $grep $opt "$pat"
     r=$?
   else
-    j=${i//\\/\\\\}
-    j=${j//|/\\|}
-    j=${j//&/\\&}
+    j=$(echo "$i" | sed -e 's/\\/\\\\/g;s/|/\\|/g;s/&/\\&/g')
     j=`printf "%s" "$j" | tr '\n' ' '`
     bzip2 -cdfq "$i" | $grep $opt "$pat" | sed "s|^|${j}:|"
     r=$?
